test/trade/actions/helpers/trade-test.js

```diff
@@ -0,0 +1,248 @@
+			buy: {
+				'order1': {
+					id: 1,
+					price: '0.45',
+					outcome: '1',
+					owner: 'owner1'
+				},
+				'order2': {
+					id: 2,
+					price: '0.45',
+					outcome: '1',
+					owner: 'owner1'
+				}
+			},
+			sell: {
+				'order3': {
+					id: 3,
+					price: '0.4',
+					outcome: '1',
+					owner: 'owner1'
+				},
+				'order4': {
+					id: 4,
+					price: '0.4',
+					outcome: '1',
+					owner: 'owner1'
+				}
+			}
+		},
+		'0x000000000000000000000000000000000scalar1': {
+			buy: {
+				'order1': {
+					id: 1,
+					price: '45',
+					outcome: '1',
+					owner: 'owner1'
+				},
+				'order2': {
+					id: 2,
+					price: '45',
+					outcome: '1',
+					owner: 'owner1'
+				}
+			},
+			sell: {
+				'order3': {
+					id: 3,
+					price: '40',
+					outcome: '1',
+					owner: 'owner1'
+				},
+				'order4': {
+					id: 4,
+					price: '40',
+					outcome: '1',
+					owner: 'owner1'
+				}
+			}
+		}
+	};
+	const store = mockStore(testState);
+	const mockAugur = { augur: { ...augur }, abi: { ...abi }, constants: { ...constants } };
+	const mockLoadBidAsks = { loadBidsAsks: () => {} };
+	sinon.stub(mockLoadBidAsks, 'loadBidsAsks', (marketID, cb) => {
+		assert.isString(marketID, `didn't pass a marketID as a string to loadBidsAsks`);
+		cb();
+		return { type: 'LOAD_BIDS_ASKS' };
+	});
+
+	sinon.stub(mockAugur.augur, 'getParticipantSharesPurchased', (marketID, userID, outcomeID, cb) => {
+		cb(10);
+	});
+
+	sinon.stub(mockAugur.augur, 'getCashBalance', (takerAddress, cb) => {
+		cb('10000.0');
+	});
+
+	sinon.stub(mockAugur.augur, 'trade', (args) => {
+		const { max_value, max_amount, trade_ids, sender, onTradeHash, onCommitSent, onCommitSuccess, onCommitFailed, onNextBlock, onTradeSent, onTradeSuccess, onTradeFailed } = args;
+		onTradeHash({ data: 'data' });
+		onCommitSent({ data: 'data' });
+		onCommitSuccess({ gasFees: '0.01450404', hash: 'testhash', timestamp: 1500000000 });
+		onCommitFailed({ code: 'error', message: 'error message' });
+		onNextBlock({ data: 'data' });
+		onTradeSent({ data: 'data' });
+		if (mockAugur.augur.trade.callCount === 1) {
+			onTradeSuccess({ sharesBought: '10', cashFromTrade: '0', unmatchedShares: '0', unmatchedCash: '0', tradingFees: '0.01', gasFees: '0.01450404', hash: 'testhash', timestamp:1500000000 });
+		} else {
+			onTradeSuccess({ sharesBought: '0', cashFromTrade: '10.00', unmatchedShares: '0', unmatchedCash: '0', tradingFees: '0.01', gasFees: '0.01450404', hash: 'testhash', timestamp:1500000000 });
+		}
+		onTradeFailed({ code: 'error', message: 'error message' });
+	});
+
+	const mockCBStatus = sinon.stub();
+	const mockCB = sinon.stub();
+
+	const helper = proxyquire('../../../../src/modules/trade/actions/helpers/trade.js', {
+		'../../../bids-asks/actions/load-bids-asks': mockLoadBidAsks,
+		'../../../../services/augurjs': mockAugur
+	});
+
+	beforeEach(() => {
+		store.clearActions();
+		mockCB.reset();
+		mockCBStatus.reset();
+	});
+
+	it('should help with sell trades', () => {
```

Comment:
The `trade` function is pretty central to trading.  So IMO we should write more than a single buy test case and a single sell test case.  Especially important is to include the relevant edge cases you can think of.  Here are a few I can think of, but this is by no means an exhaustive list:
    - What if there are no matching trade IDs (was filled at the last second before `trade` was called)?
    - What if there are a *lot* of trade IDs, maybe 50 or 100?
    - Right now trade IDs are chunked in batches of 3, so probably worth at minimum testing a number of trade IDs that is a multiple of 3 and a number that isn't.
    - Does `trade` cope gracefully with stringified number inputs?  BigNumber inputs?  JS number inputs?  (It should handle all 3; there should be test cases for each.)
    - How about if some/all of the inputs are negative?
    - How about if some/all of the inputs are null or undefined?  (If it throws an error, test to make sure it's the exact error that is expected for that test case.)
    - What happens if the `onFailed` handlers are invoked?  Does `trade` deal gracefully with these?  How is the transaction object updated if `onFailed` is called?
    - There are 4 different conditions in [async.until](https://github.com/caolan/async/blob/v1.5.2/README.md#until) that can cause the loop to terminate.  There should be at least 1 test case that covers each of these conditions.
    - What happens if `augur.getParticipantSharesPurchased` or `augur.getCashBalance` return a value that is different than expected (perhaps the user had other actions running in parallel with the trade)?  Does `trade` respond appropriately, or does it still try and use the (now outdated) values?